showcase/app/templates/components/form/super-select.hbs

```diff
@@ -0,0 +1,1020 @@
+{{! @glint-nocheck: not typesafe yet }}
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+{{page-title "Form::SuperSelect Components"}}
+
+<BasicDropdownWormhole />
+
+<Shw::Text::H1>Form::SuperSelect</Shw::Text::H1>
+
+<section data-test-percy>
+  <Shw::Text::H2>Form::SuperSelect::Single::Base</Shw::Text::H2>
+
+  <Shw::Text::H3>Interaction status</Shw::Text::H3>
+
+  <Shw::Grid @columns={{3}} as |SG|>
+    <SG.Item @label="Default">
+      <Hds::Form::SuperSelect::Single::Base
+        @onChange={{this.noop}}
+        @options={{@model.OPTIONS}}
+        @ariaLabel="Label"
+        as |option|
+      >
+        {{option}}
+      </Hds::Form::SuperSelect::Single::Base>
+    </SG.Item>
+    <SG.Item @label="With placeholder">
+      <Hds::Form::SuperSelect::Single::Base
+        @onChange={{this.noop}}
+        @options={{@model.OPTIONS}}
+        @placeholder="Placeholder text"
+        @ariaLabel="Label"
+        as |option|
+      >
+        {{option}}
+      </Hds::Form::SuperSelect::Single::Base>
+    </SG.Item>
+    <SG.Item @label="Selected">
+      <Hds::Form::SuperSelect::Single::Base
+        @onChange={{fn (mut @model.SELECTED_OPTION)}}
+        @options={{@model.OPTIONS}}
+        @selected={{@model.SELECTED_OPTION}}
+        @ariaLabel="Label"
+        as |option|
+      >
+        {{option}}
+      </Hds::Form::SuperSelect::Single::Base>
+    </SG.Item>
+  </Shw::Grid>
+
+  <Shw::Divider @level="2" />
+
+  <Shw::Text::H3>States</Shw::Text::H3>
+
+  <Shw::Grid @columns={{3}} as |SG|>
+    {{#let (array "base" "invalid" "disabled") as |variants|}}
+      {{#each @model.STATES as |state|}}
+        <SG.Item>
+          <Shw::Flex @direction="column" as |SF|>
+            {{#each variants as |variant|}}
+              {{! template-lint-disable simple-unless }}
+              {{#unless (and (eq variant "disabled") (eq state "focus"))}}
+                <SF.Item @label="{{capitalize variant}} / {{capitalize state}}">
+                  <Hds::Form::SuperSelect::Single::Base
+                    @onChange={{fn (mut @model.SELECTED_OPTION)}}
+                    @options={{@model.OPTIONS}}
+                    @selected={{@model.SELECTED_OPTION}}
+                    @disabled={{if (eq variant "disabled") true}}
+                    @isInvalid={{if (eq variant "invalid") true}}
+                    @ariaLabel="Label"
+                    mock-state-value={{state}}
```

Comment:
I _think_ the mock states are not working here? 🤔
<img width="998" alt="screenshot_3768" src="https://github.com/hashicorp/design-system/assets/686239/a8d2f59b-0add-4d5f-b10c-d36bbbcdb819">

Reply:
I broked this when [applying suggestions from code review](https://github.com/hashicorp/design-system/commit/d2c913ca833bb9b48e7632e74a28f6231782c95d) – our script (`mock-state-value={{state}}`) doesn't seem to be working on the PowerSelect (likely a race or an override when writing values for `class`), which is probably why @KristinLBradley applied the `class="mock-{{state}}"` directly

Reply:
OK, my bad. Didn't know PowerSelect did this (likely re-rendering the component, or part of it, after the page has complete its rendering).

Reply:
Fixed in [0a3221b](https://github.com/hashicorp/design-system/pull/2022/commits/0a3221be46c3798dc9c6a9576db83808a35b8940)